.github/workflows/nodejs-setup.yml

```diff
@@ -0,0 +1,30 @@
+name: Node.js Setup and Test
```

Comment:
Refactor into a shared file so we can be sure the checks are always run.

Reply:
smart!